src/m365/teams/commands/meeting/meeting-transcript-list.ts

```diff
@@ -0,0 +1,115 @@
+import auth from '../../../../Auth';
+import { Logger } from '../../../../cli/Logger';
+import GlobalOptions from '../../../../GlobalOptions';
+import GraphCommand from "../../../base/GraphCommand";
+import commands from '../../commands';
+import { odata } from '../../../../utils/odata';
+import { validation } from '../../../../utils/validation';
+import { accessToken } from '../../../../utils/accessToken';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  userId?: string;
+  userName?: string;
+  email?: string;
+  meetingId: string;
+}
+
+class TeamsMeetingTranscriptListCommand extends GraphCommand {
+  public get name(): string {
+    return commands.MEETING_TRANSCRIPT_LIST;
+  }
+
+  public get description(): string {
+    return 'Lists all transcripts for a given meeting';
+  }
+
+  public defaultProperties(): string[] | undefined {
+    return ['id', 'createdDateTime'];
+  }
+
+  constructor() {
+    super();
+
+    this.#initTelemetry();
+    this.#initOptions();
+    this.#initValidators();
+  }
+
+  #initTelemetry(): void {
+    this.telemetry.push((args: CommandArgs) => {
+      Object.assign(this.telemetryProperties, {
+        userId: typeof args.options.userId !== 'undefined',
+        userName: typeof args.options.userName !== 'undefined',
+        email: typeof args.options.email !== 'undefined'
+      });
+    });
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      {
+        option: '-u, --userId [userId]'
+      },
+      {
+        option: '-n, --userName [userName]'
+      },
+      {
+        option: '--email [email]'
+      },
+      {
+        option: '-m, --meetingId <meetingId>'
+      }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        if (args.options.userId && !validation.isValidGuid(args.options.userId)) {
+          return `${args.options.userId} is not a valid Guid`;
+        }
+
+        return true;
+      }
+    );
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    const isAppOnlyAuth: boolean = accessToken.isAppOnlyAccessToken(auth.service.accessTokens[this.resource].accessToken);
+
+    try {
+      if (this.verbose) {
+        logger.logToStderr(`Retrieving transcript list for the given meeting...`);
+      }
+
+      let requestUrl = `${this.resource}/v1.0/`;
+      if (isAppOnlyAuth) {
+        requestUrl += 'users/';
+        if (args.options.userId) {
+          requestUrl += args.options.userId;
+        }
+        else {
+          this.handleError(`The option 'userId' is required when using the app-only authentication`);
+        }
+      }
+      else {
+        requestUrl += `me`;
+      }
```

Comment:
We still need to fix this part. I verified how the command meeting-list.ts was created and they are also performing a check based on isAppOnlyAuth. 

You should be able to reuse the logic from there. We only need to retrieve the user ID when the email option is passed. 

Then we should be fine.